### PR TITLE
Adjust click area for series title inline edit

### PR DESCRIPTION
**Pull request by Coenego**
Keep clicking accidentally on the hotspot, with is 80% of the width I think at the monent.

We should optimise this so that the hotspot for invoking the inline edit should only be the length of the title string.

timetable_administration

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @bp323 for review

File: shared/gh/js/views/gh.admin-series-title.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            // Add an `editing` class when the input field is being focussed
+            $('input', this).on('focus', function() {
+                $(original).addClass('editing');
+            });
+
+            // Remove the `editing` class when the input field loses its focus
+            $('input', this).on('focusout', function() {
+                setTimeout(function() {
+                    $(original).removeClass('editing');
+                }, 750);
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-series-title.js]: Do we really need the timeout? A problem that popped up is that reinitialising the editable field before the class has been removed results in a strange looking input field:

screen shot 2015-03-17 at 14 25 14

**Comment by bp323** [comment]
Reassigning to @Coenego for a fix

**Comment by Coenego** [comment]
Follow-up done, reassigning to @bp323 for review

File: apps/timetable/admin/ui/css/admin.css [hunk 2]
```diff
@@ -396,10 +396,15 @@
 
 #gh-batch-edit-container h1 {
     cursor: pointer;
+    display: inline-block;
     font-size: 25px;
     font-weight: 400;
     height: 40px;
```
Comment (bp323) [inline apps/timetable/admin/ui/css/admin.css]: Can you change this to min-height

Comment (bp323) [inline apps/timetable/admin/ui/css/admin.css]: And add a `max-width: 80%`

Comment (bp323) [inline apps/timetable/admin/ui/css/admin.css]: That'll fix the problem of multiple lines pushing the h1 to the next line

**Comment by bp323** [comment]
Back to @Coenego for minor followup

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 3]
```diff
@@ -516,6 +516,10 @@ define(['gh.core', 'gh.constants', 'gh.utils', 'moment', 'gh.calendar', 'gh.admi
      * @private
      */
     var editableSeriesTitleSubmitted = function(value, editableField) {
+
+        // Remove the editing class from the jeditable input field
+        $('h1.editing').removeClass('editing');
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit.js]: Can you make this more specific to the title so only one selector will ever match this?

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 4]
```diff
@@ -633,15 +637,21 @@ define(['gh.core', 'gh.constants', 'gh.utils', 'moment', 'gh.calendar', 'gh.admi
...
-            'placeholder': '',
-            'select' : true,
-            'submit': '<button type="submit" class="btn btn-default">Save</button>'
+            'placeholder': 'Click to add a title for this series',
+            'select': true,
+            'type': 'series-title',
+            'submit': '<button type="submit" class="btn btn-default">Save</button>',
+            'callback': function(value, settings) {
+                // Remove the `editing` class when the input field loses its focus
+                $('h1.editing').removeClass('editing');
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit.js]: Here as well

**Comment by Coenego** [comment]
Follow-up done, reassigning to @bp323

**Comment by bp323** [comment]
Verified and merged.

**Pull request merged**